aarch64/kernel/nano_core/context_switch.rs

```diff
@@ -41,48 +40,25 @@ pub fn switch_to_task(new_stack: usize) {
 /// `context_switch_regular` with that new stack.
 pub fn create_stack(
     page_table: &mut PageTable,
-    start_address: usize,
+    start_address: extern "C" fn(),
     pages: usize,
 ) -> Result<(MappedPages, usize), &'static str> {
-    // x30 stores the return address
-    // it's used by the `ret` instruction
-    let artificial_ctx_src = [
-        // [lower address]
-        // [returned stack pointer]
-
-                    0, 0, //  x2,  x3,
-                    0, 0, //  x4,  x5,
-                    0, 0, //  x6,  x7,
-                    0, 0, //  x8,  x9,
-                    0, 0, // x10, x11,
-                    0, 0, // x12, x13,
-                    0, 0, // x14, x15,
-                    0, 0, // x16, x17,
-                    0, 0, // x18, x19,
-                    0, 0, // x20, x21,
-                    0, 0, // x22, x23,
-                    0, 0, // x24, x25,
-                    0, 0, // x26, x27,
-                    0, 0, // x28, x29,
-        start_address, 0, // x30, <nothing; kept for alignment>
-
-        // [higher address]
-        // [top of stack]
-    ];
+    let artificial_ctx = ContextRegular::new(start_address as *const () as _);
+    let artificial_ctx_src = artificial_ctx.as_bytes();
```

Comment:
we don't need `as_bytes()` here -- we can directly write the object itself to the region of memory as long as it implements `FromBytes`. Not only is that faster, but it's also more readable in terms of the code's intent.

Look at how we do it in `spawn::setup_context_trampoline()`